### PR TITLE
Improve Dropdown accessibility

### PR DESCRIPTION
**Pull request by SevenOutman**
## More ARIA attributes applied

- Add `aria-controls` and `aria-labelledby` relation between the toggle button and the menu.
- Add `aria-haspopup`, `aria-controls` and `aria-labelledby` relation between the submenu and the menuitem that controls it.
- Add `aria-expanded` state on menu controls.
- Fix incorrect role assigned to the menuitems that controls submenus. (`menu` -> `menuitem`)

### When menu button is focused (menu is closed)

| Keydown | Action
| --------- | ------
| <kbd>Enter</kbd> | Open the menu and move focus to the first item
| <kbd>Space</kbd> | Open the menu and move focus to the first item
| <kbd>ArrowDown</kbd> | Open the menu and move focus to the first item

### When menu is focused (menu is open)

| Keydown | Action
| --------- | ------
| <kbd>Enter</kbd> | When focus is on a menuitem that has a submenu, opens the submenu and places focus on its first item. Otherwise, activates the item and closes the menu.
| <kbd>Space</kbd> | When focus is on a menuitem that has a submenu, opens the submenu and places focus on its first item. Otherwise, activates the item and closes the menu.
| <kbd>ArrowDown</kbd> | Move focus to the next item
| <kbd>ArrowUp</kbd> | Move focus to the previous item
| <kbd>ArrowRight</kbd> | When focus is in a menu and on a menuitem that has a submenu, opens the submenu and places focus on its first item.
| <kbd>ArrowLeft</kbd> | When focus is in a submenu of an item in a menu, closes the submenu and returns focus to the parent menuitem.
| <kbd>Home</kbd> | Move focus to the first item
| <kbd>End</kbd> | Move focus to the last item
| <kbd>Escape</kbd> | Close the menu and move focus to the context (menu button or menuitem that opens this submenu)

### Reference

- https://www.w3.org/TR/wai-aria-practices-1.2/#menubutton
- https://www.w3.org/TR/wai-aria-practices-1.2/#kbd_general_within
- https://www.w3.org/TR/wai-aria-practices-1.2/#kbd_focus_activedescendant

## Changes

- Deprecate `pullLeft` prop on `<Dropdown.Menu>`. Submenu always points the reading flow direction.

### CODE REVIEW

File: src/Dropdown/Menu.tsx [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+      'aria-activedescendant': menuControl.items[menuControl.activeItemIndex]?.id
+    };
+
+    /**
+     * Keyboard interaction on menu
+     * @see https://www.w3.org/TR/wai-aria-practices-1.2/#keyboard-interaction-12
+     */
+    const handleKeydown = useCallback(
+      (e: React.KeyboardEvent<HTMLUListElement>) => {
+        const activeItem = menuControl.items[menuControl.activeItemIndex];
```
Comment (simonguo) [inline src/Dropdown/Menu.tsx:110]: menuControl.items 中的 Item 需要过滤掉被禁用的节点。 在键盘 up ,down 操作的时候可以跳过这些选项。

Comment (simonguo) [reply src/Dropdown/Menu.tsx:110]: 还需要过滤掉 item 是 separator 的选项。
image

File: src/Dropdown/MenuItem.tsx [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+          className={classes}
+        >
+          <div
+            ref={menuitemRef as any}
+            id={menuitemId}
+            className={prefix`toggle`}
+            aria-controls={(rest as any).id}
+            tabIndex={-1}
+            {...menuitemAriaAttributes}
+            {...(menuitemEventHandlers as any)}
```
Comment (simonguo) [inline src/Dropdown/MenuItem.tsx:324]: submenu: 鼠标操作 mouseOver 到文字可以展开子节点，但在其他区域则无法展开。
你看是把事件绑定在 li 上，还是绑定在 div 上？  或者调整 div 的样式。

**Comment by simonguo** [comment]
## 汇总相关组件问题:
- Sidenav 中相关 submenu 样式有问题
image
- Sidenav activeKey 对应的高亮选项会出现多个。
## 可优化的点
- Sidenav 中键盘操作不能在多个 dropdown 中切换， 
- 在 dropdown toggle上回车的时候可以撑开折叠 submenu

**Comment by SevenOutman** [comment]
> * Sidenav 中键盘操作不能在多个 dropdown 中切换，
> * 在 dropdown toggle上回车的时候可以撑开折叠 submenu

`<Sidenav>` 对应的 ARIA 组件应该是 `tree` ([参考此示例](https://www.w3.org/TR/wai-aria-practices-1.1/examples/treeview/treeview-2/treeview-2a.html))，其键盘操作应当另外实现。

**Comment by simonguo** [comment]
image
点击包含子节点的第二级节点，不能展开和折叠菜单。 在没有设计出更好的API的情况下先保证之前功能的完整性。

**Comment by SevenOutman** [comment]
> 点击包含子节点的第二级节点，不能展开和折叠菜单。

已在 #1744 中修复

**Pull request merged**